src/test/scala/mesosphere/marathon/tasks/InstanceOpFactoryImplTest.scala

```diff
@@ -156,16 +155,15 @@ class InstanceOpFactoryImplTest extends UnitTest with Inside {
       val localVolumeIdUnwanted = LocalVolumeId(app.id, "persistent-volume-unwanted", "uuidUnwanted")
       val localVolumeIdMatch = LocalVolumeId(app.id, "persistent-volume", "uuidMatch")
       val reservedInstance = f.residentReservedInstance(app.id, localVolumeIdMatch)
-      val (reservedTaskId, _) = reservedInstance.tasksMap.head
+      val reservedTaskId = Task.Id.forInstanceId(reservedInstance.instanceId, None)
       val offer = f.offerWithVolumes(
         reservedTaskId, localVolumeIdLaunched, localVolumeIdUnwanted, localVolumeIdMatch
       )
       val runningInstances = Instance.instancesById(Seq(
-        f.residentLaunchedInstance(app.id, localVolumeIdLaunched),
-        reservedInstance))
+        f.residentLaunchedInstance(app.id, localVolumeIdLaunched)))
```

Comment:
Why did you remove the second instance?

Reply:
because reserved instance is not running instance now... (it might have been considered as one before)

Reply:
The `reservedInstance` needs to be contained in the map. The signature of the Request is
```
  case class Request(
      runSpec: RunSpec,
      offer: Mesos.Offer,
      instanceMap: Map[Instance.Id, Instance],
      scheduledInstances: Iterable[Instance],
      localRegion: Option[Region] = None)
```

Where the instanceMap is supposed to contain **all** reachable instances. Those are used e.g. to match constraints. The `TaskLauncherActor` does this:
```
val matchRequest = InstanceOpFactory.Request(runSpec, offer, reachableInstances, scheduledInstances, localRegion())
```

while the logic there seems to be broken:
```
val reachableInstances = instanceMap.filterNotAs{
    case (_, instance) => instance.state.condition.isLost || instance.isScheduled
}
```

This should **not** exclude instances w/ reservation (i.e. agentInfo). Otherwise we could create additional reservations on an agent that already has one (or multiple!), and we'd never be able to launch tasks for more than one of them.

Reply:
hmm I don't think that's true but it was hard to tell - I adjusted the names here a bit.
This is *SCHEDULED* instance with reservation, so it should not be among running instances